src/main/webapp/babel-src/feedBackBox.js

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import Dialog from 'material-ui/Dialog';
+import FlatButton from 'material-ui/FlatButton';
+import TextField from 'material-ui/TextField';
+import Snackbar from 'material-ui/Snackbar';
+import {FEEDBACK_CHAR_LIMIT, FEEDBACK_SNACKBAR_AUTO_HIDE_DURATION, UI_STRINGS, INLINE_STYLES} from './util';
+
+export class FeedBackBox extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            charCtr : 0,
+            errorMsg : '',
+            feedbackMsg: '',

```

Comment:
considering that this feedbackMsg variable is not being used in the render function (there is no UI element whose contents are derived from the value of this variable), we can probably get better performance if it is changed to be a class attribute rather than being in the state object (i.e. this.feedbackMsg instead of this.state.feedbackMsg). This is because every time the state is updated, React will trigger the render function again. This would also mean that you would assign new values to it directly like this.feedbackMsg = 'somevalue' instead of using the setState() function

Reply:
done